pkg/services/oauthtoken/oauth_token.go

```diff
@@ -157,21 +186,12 @@ func (o *Service) tryGetOrRefreshAccessToken(ctx context.Context, usr *models.Us
 	}
 	ctx = context.WithValue(ctx, oauth2.HTTPClient, client)
 
-	persistedToken := &oauth2.Token{
-		AccessToken:  usr.OAuthAccessToken,
-		Expiry:       usr.OAuthExpiry,
-		RefreshToken: usr.OAuthRefreshToken,
-		TokenType:    usr.OAuthTokenType,
-	}
-
-	if usr.OAuthIdToken != "" {
-		persistedToken = persistedToken.WithExtra(map[string]interface{}{"id_token": usr.OAuthIdToken})
-	}
+	persistedToken := buildOAuthTokenFromAuthInfo(usr)
 
 	// TokenSource handles refreshing the token if it has expired
 	token, err := connect.TokenSource(ctx, persistedToken).Token()
 	if err != nil {
-		logger.Error("failed to retrieve oauth access token", "provider", usr.AuthModule, "userId", usr.UserId, "error", err)
+		logger.Warn("failed to retrieve oauth access token", "provider", usr.AuthModule, "userId", usr.UserId, "error", err)
```

Comment:
`tryGetOrRefreshAccessToken` is intended to be called after the OAuth RT existence is verified. However, currently this is not the case, because currently the RT existence check (`isRefreshableOAuthProvider`) is outside of this function, so it might be a good idea to move the `isRefreshableOAuthProvider` to this function and log an error here. 

What do you think?

Reply:
yeah, now with the main problem (`oauth2: token expired and refresh token is not set`) fixed it should spam the logs less I hope. 

We'll see which errors remain, if they're expected errors I'll downgrade this to a warn after